index.d.ts

```diff
@@ -0,0 +1,16 @@
+
+import type MarkdownIt from 'markdown-it';

```

Comment:
I'm not sure that's necessary. Anyone consuming these types should also have `@types/markdown-it` installed, even if that package isn't a direct dependency of this project. Further, I think if the type names have changed or aren't matching, then that likely is an indication of a greater API change that would render the whole plugin incompatible anyways. Might be good to add `@types/markdown-it` as a dev dependency though.

Reply:
I am fairly confident in that opinion, but I agree, it is an opinion, maybe have a look into how other plugins do it? My guess would be that at the very least the markdown it would then be part of the peerDependencies. By choosing to untie the dependencies, it should just make things easier to maintain and care for.

Reply:
Looks like adding the types and potentially also `markdown-it` itself to the peer dependencies may be the way to go. Here's [`markdown-it-anchor`](https://github.com/valeriangalliat/markdown-it-anchor/blob/master/package.json#L43-L46) for example.

Reply:
@michaelhthomas Thank you for looking it up. Would you mind adding the peer-dependencies?